src/Data/List/Lazy.purs

```diff
@@ -643,14 +645,26 @@ transpose xs =
           (x : mapMaybe head xss) : transpose (xs : mapMaybe tail xss)
 
 --------------------------------------------------------------------------------
+-- Infinite list inspection ----------------------------------------------------
+--------------------------------------------------------------------------------
+
+-- | Like `show`, but only shows the first n elements. Useful for potentially
+-- | infinite lists.
+showPrefix :: forall a. (Show a) => Int -> List a -> String
```

Comment:
I think if users need this, they can just write `show <<< take n`. I do this frequently in Haskell, for example.